samples/csharp/getting-started/Ranking_PersonalizedSort/PersonalizedRanking/Common/ConsoleHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.ML;
+using Microsoft.ML.Data;
+using PersonalizedRanking.DataStructures;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PersonalizedRanking.Common
+{
+    public class ConsoleHelper
+    {
+        // To evaluate the accuracy of the model's predicted rankings, prints out the Discounted Cumulative Gain and Normalized Discounted Cumulative Gain for hotel search queries.
+        public static void EvaluateMetrics(MLContext mlContext, IDataView scoredData)
+        {
+            // Evaluate the metrics for the data using NDCG; by default, metrics for the up to 3 search results in the query are reported (e.g. NDCG@3).
+            RankingMetrics metrics = mlContext.Ranking.Evaluate(scoredData);
+
+            Console.WriteLine($"DCG: {string.Join(", ", metrics.DiscountedCumulativeGains.Select((d, i) => $"@{i + 1}:{d:F4}").ToArray())}");
+
+            Console.WriteLine($"NDCG: {string.Join(", ", metrics.NormalizedDiscountedCumulativeGains.Select((d, i) => $"@{i + 1}:{d:F4}").ToArray())}");
+        }
+
+        // Performs evaluation with the truncation level set up to 10 hotel search results within a query.
+        // This is a temporary workaround for this issue: https://github.com/dotnet/machinelearning/issues/2728.
+        public static void EvaluateMetrics(MLContext mlContext, IDataView scoredData, int truncationLevel)
+        {
+            if (truncationLevel < 1 || truncationLevel > 10)
+            {
+                throw new InvalidOperationException("Currently metrics are only supported for 1 to 10 truncation levels.");
```

Comment:
Is there something causing only support up to n=10?

Reply:
I don't know for sure - I actually had tried setting this value to greater than 10 (specifically 38 since this was what was used for the Kaggle contest).   It turns out that the code has explicit checks for greater than 10 and throws an exception (which is why I'm also doing this check).  And, even more confusing is that there are comments in the ML .NET code indicating that a value up to 100 should be allowed (but with a ? mark behind it).  I talked with @eerhardt about this and he suspects that there must be a reason that they've limited it to 10.  I plan to log an issue for this.

Reply:
I don't expect there should be a reason to limit the number of slots used in the DCG/NDCG calculation.